### PR TITLE
Hash table check little optimization

### PR DESCRIPTION
**Pull request by zhanglistar**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Hash table check little optimization.

### CODE REVIEW

**Review by evillique — APPROVED** [review]

**Comment by evillique** [comment]
This change brakes the ClangTidy build: [report](https://s3.amazonaws.com/clickhouse-builds/43853/b7537e8ffb28e9aecdc801ffd464409f5a310610/binary_tidy/build_log.log) and does not really show the performance improvement.

File: src/Common/HashTable/HashTable.h [hunk 1]
```diff
@@ -70,7 +70,7 @@ namespace ZeroTraits
 {
 
 template <typename T>
-bool check(const T x) { return x == T{}; }
+bool check(const T & x) { return x == T{}; }
```
Comment (alexey-milovidov) [inline src/Common/HashTable/HashTable.h:73]: This change should not have any performance effects.

Comment (zhanglistar) [reply src/Common/HashTable/HashTable.h:73]: @alexey-milovidov Could you please explain why? Thanks in advance.

Comment (alexey-milovidov) [reply src/Common/HashTable/HashTable.h:73]: If the function body is in the same translation unit, the compiler can change the argument to pass-by-value.
Moreover, the function is inlined, and there will be no argument passing, neither by reference nor in registers.

**Pull request closed without merging**